webapp/src/context/auth-context.tsx

```diff
@@ -0,0 +1,40 @@
+import { createContext, ReactNode, useEffect, useState } from 'react';
+import { User } from '../shared/generated';
+import { authApi, usersApi } from '../shared/services/ApiService';
+
+interface AuthContextType {
+  user: User | null;
+  isLoading: boolean;
+  logout: (callback: VoidFunction) => void;
+}
+
+export const AuthContext = createContext<AuthContextType>(null!);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    setIsLoading(true);
+    usersApi
+      .userControllerGetCurrentUser()
+      .then((currentUser) => setUser(currentUser))
+      .catch(() => setUser(null))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const logout = async (callback: VoidFunction) => {
```

Comment:
Twenty lines of code, two ways of writing async functions.

Reply:
I've changed it to use the `useData` hook. Please let me know what you think.

Reply:
I think it looks good.